### ISSUE TITLE
Conda only dependency

### ISSUE THREAD

**Issue by churnikov** [dojeda/poetry2conda#11]
Hello!

I think this is a great tool, but I think there is one package that stops me from using it. I'm talking about [rdkit](https://github.com/rdkit/rdkit).

You see, it's impossible to install it as a git dependency and the simplest way to install it is via Conda.

Can I use your tool in this case? And how 😊?)

**Comment by dojeda** [comment]
I don't think you can use poetry or poetry2conda for this particular package. Poetry relies on packages that are distributed through PyPI. I don't really understand what you are trying to achieve, given that your dependency is a "pure conda" package (which is regrettable).

**Comment by damienallen** [comment]
I'm facing a different use case that could also be described as a "conda only dependency". We use poetry for all our env management but need conda for our pyinstaller builds. Thus, we want pyinstaller in our conda env but not in our poetry envs.

Not sure if this really helps OP but maybe someone passing through in the future. We simply used the "extras" functionality in pyproject.toml:

```
[tool.poetry.dependencies]
...
pyinstaller = { version = "3.6", optional = true }

[tool.poetry.extras]
conda-only = ["pyinstaller"]
```

Then we call poetry2conda with the extras flag: `poetry2conda --extras conda-only pyproject.toml conda.yml `

@dojeda Really appreciate this tool!

**Comment by frankier** [comment]
I was able to use the above trick to install a package not on pypi at all:

```
[tool.poetry.dev-dependencies]
dummypkg = { git="https://github.com/bjodah/dummypkg.git", tag="master", optional = true }
...

[tool.poetry.extras]
conda-only = ["dummypkg"]

[tool.poetry2conda.dependencies]
dummypkg = { name="cuml", version = "0.18", channel = "rapidsai" }
...
```

**Comment by chestergan** [comment]
> I'm facing a different use case that could also be described as a "conda only dependency". We use poetry for all our env management but need conda for our pyinstaller builds. Thus, we want pyinstaller in our conda env but not in our poetry envs.
> 
> Not sure if this really helps OP but maybe someone passing through in the future. We simply used the "extras" functionality in pyproject.toml:
> 
> ```
> [tool.poetry.dependencies]
> ...
> pyinstaller = { version = "3.6", optional = true }
> 
> [tool.poetry.extras]
> conda-only = ["pyinstaller"]
> ```
> 
> Then we call poetry2conda with the extras flag: `poetry2conda --extras conda-only pyproject.toml conda.yml `
> 
> @dojeda Really appreciate this tool!

@dojeda  This should be added to the package's descriptions; there is not a manual for all the possible yaml headers & configurations!

Moreover, I think `poetry update` should have an additional option, similar to `--no-dev`, to exclude installations & updates on extra dependencies.

**Comment by chestergan** [comment]
> I'm facing a different use case that could also be described as a "conda only dependency". We use poetry for all our env management but need conda for our pyinstaller builds. Thus, we want pyinstaller in our conda env but not in our poetry envs.
> 
> Not sure if this really helps OP but maybe someone passing through in the future. We simply used the "extras" functionality in pyproject.toml:
> 
> ```
> [tool.poetry.dependencies]
> ...
> pyinstaller = { version = "3.6", optional = true }
> 
> [tool.poetry.extras]
> conda-only = ["pyinstaller"]
> ```
> 
> Then we call poetry2conda with the extras flag: `poetry2conda --extras conda-only pyproject.toml conda.yml `
> 
> @dojeda Really appreciate this tool!

How do you run `poetry update` on the .toml file successfully? `poetry update` still encounters the SolverProblemError for the conda-specific package `nomkl`:
![image](https://user-images.githubusercontent.com/55651551/119222488-2f807b00-bb27-11eb-9bd0-2b0dbc2b3c4d.png)

**Comment by chestergan** [comment]
> I was able to use the above trick to install a package not on pypi at all:
> 
> ```
> [tool.poetry.dev-dependencies]
> dummypkg = { git="https://github.com/bjodah/dummypkg.git", tag="master", optional = true }
> ...
> 
> [tool.poetry.extras]
> conda-only = ["dummypkg"]
> 
> [tool.poetry2conda.dependencies]
> dummypkg = { name="cuml", version = "0.18", channel = "rapidsai" }
> ...
> ```

This works when there is only one conda-specific package, but what if there is more? How many dummy packages do I need to use?
